enhancements/run-bundle.md

```diff
@@ -0,0 +1,263 @@
+ 1. Create pod using `registryImage`, set container command (entrypoint) to:
+    ```
+    /bin/bash -c ‘ \
+      /bin/mkdir -p /database && \
+      /bin/opm registry add   -d /database/index.db -b {.BundleImage} && \
+      /bin/opm registry serve -d /database/index.db’
+    ```
+ 1. If pod fails, clean it up, and error out with message of unsupported
+    index image
+ 1. Create GRPC CatalogSource, pointing to <pod-name>.<namespace>:50051
+ 1. Create OperatorGroup, only if it does not exist, since `--install-mode` was
+    not supplied, defaults to `AllNamespaces`.
+ 1. Create Subscription
+ 1. Operator installed into the default namespace defined by KUBECONFIG
+ 1. Verify operator is installed.
+ 1. Exit with success (leaving operator running)
+
+#### Complex Example
+
+Now let's look at one of the more complex examples:
+
+```
+operator-sdk run bundle <bundle-image> --namespace foo --install-mode SingleNamespace=bar
+```
+
+ 1. Since `--index-image` was not provided, assign variable `registryImage`
+    equal to `quay.io/operator-framework/upstream-registry-builder`
+ 1. Create pod using `registryImage`, set container command (entrypoint) to:
+    ```
+    /bin/bash -c ‘ \
+      /bin/mkdir -p /database && \
+      /bin/opm registry add   -d /database/index.db -b {.BundleImage} && \
+      /bin/opm registry serve -d /database/index.db’
+    ```
+ 1. If pod fails, clean it up, and error out with message of unsupported
+    index image
+ 1. Create GRPC CatalogSource, pointing to <pod-name>.<namespace>:50051
+ 1. Creates a `SingleNamespace` OperatorGroup watching namespace `bar`.
+ 1. Create Subscription
+ 1. Operator installed into the default namespace `foo`
+ 1. Verify operator is installed.
+ 1. Exit with success (leaving operator running)
+
+#### Prototype Links
+
+ * PR upstream-opm-builder.Dockerfile: add bash and ca-certificates
+   https://github.com/operator-framework/operator-registry/pull/320
+ * Examples setup script:
+   https://gist.github.com/joelanford/f7a7d89977638547b410db3eafc35a62
+ * Modified upstream-opm-builder:
+   https://github.com/joelanford/operator-registry/blob/9f8b6b19709a333ede683b89ac0fda1abf9b14b8/upstream-opm-builder.Dockerfile#L16-L17
+ * Bundle runner Helm chart:
+   https://github.com/joelanford/bundle-runner
+ * Static validation sync:
+   https://docs.google.com/document/d/1Lf9foKtObj1ACk67a_0BNn7rupWAHEejZ6vynJggBCw/edit#
+ * Bundle format:
+   https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md
+
+### Risks and Mitigations
+
+If we do not get the modified
+[upstream-opm-builder](https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md)
+referenced above the following assumptions will not work:
+
+  1. if image doesn't have bash, we can't modify the entrypoint
+  1. if opm is building index images without bash, then there will be no index
+     images we support
```

Comment:
If we want to support redhat supported indexes, we should also check to see what their base image is. I know they are definitely doing some replacement there. @gallettilance